src/common/schema/application.js

```diff
@@ -390,11 +390,11 @@ module.exports = {
     'dischargeType'
   ],
   dependencies: {
-    veteranGrossIncome: ['understandsFinancialDisclosure'],
-    veteranNetIncome: ['understandsFinancialDisclosure'],
-    veteranOtherIncome: ['understandsFinancialDisclosure'],
-    spouseGrossIncome: ['understandsFinancialDisclosure'],
-    spouseNetIncome: ['understandsFinancialDisclosure'],
-    spouseOtherIncome: ['understandsFinancialDisclosure']
+    veteranGrossIncome: { properties: { understandsFinancialDisclosure: { 'enum': [true] } } },
```

Comment:
Sorry, this is kind of mind-melting...

I think the issue is that we have several fields that depend not only on `understandsFinancialDisclosure`, but also on other factors (for example, `understandsFinancialDisclosure` only needs `spouseGrossIncome` if `maritalStatus !== 'Never Married'`).

Or another way to put it...  `spouseGrossIncome`, if filled out, needs `understandsFinancialDisclosure` to be true, but `understandsFinancialDisclosure` doesn't necessarily need  `spouseGrossIncome` to be filled out.

We could build out the whole dependency tree, but that's not really how we're using JSONSchema... it's mostly there as a quick sanity check before we send the application off to ES.

My recommendation is that we keep going down the path we're already on-- Let's use JSONSchema as a basic rubric and not necessarily try to duplicate all of ES's schema.